sematic/container_images.py

```diff
@@ -19,28 +20,46 @@ def has_container_image() -> bool:
     True if an image can be found, False otherwise.
     """
     try:
-        get_image_uri()
+        get_image_uris()
         return True
-    except MissingCloudImage:
+    except MissingContainerImage:
         return False
 
 
-def get_image_uri() -> str:
+def get_image_uris() -> Dict[str, str]:
     """Get the URI of the docker image associated with this execution.
 
     Returns
     -------
     The URI of the image to be used in this execution.
     """
     if CONTAINER_IMAGE_ENV_VAR in os.environ:
-        return os.environ[CONTAINER_IMAGE_ENV_VAR]
+        return {CONTAINER_IMAGE_ENV_VAR: os.environ[CONTAINER_IMAGE_ENV_VAR]}
 
-    file_path = "{}_push_at_build.uri".format(os.path.splitext(__main__.__file__)[0])
-    if not pathlib.Path(file_path).exists():
-        raise MissingCloudImage(
+    tagged_uris_map = {}
+
+    dir_path, file_name = os.path.split(__main__.__file__)
+
+    image_file_regex = r"{}_(.*)_push_at_build.uri".format(
```

Comment:
This matching problem is ambiguous.

`(.*)` is greedy and will match as much as possible; `my_target_my_tag_push_at_build.uri` will be captured as (`my`, `target_my_tag`).

`(.*?)` is non-greedy and will match as little as possible; `my_target_my_tag_push_at_build.uri` will be captured as (`my_target_my`, `tag`).

The bazel rule URI generation should avoid this by using two underscores to separate the target and tag, and should validate target and tag values to not have double underscores.

Reply:
```
>>> re.findall(r"my_target_(.*)_push_at_build.uri", "my_target_my_tag_push_at_build.uri")
['my_tag']
>>> re.findall(r"my_target_(.*?)_push_at_build.uri", "my_target_my_tag_push_at_build.uri")
['my_tag']
```
Seems fine either way.